README.md

```diff
@@ -55,6 +55,33 @@ Documentation will be generated in target/doc/...
 Most documentation is taken from NaCl, with minor modification where the API
 differs between the C and Rust versions.
 
+#### Building and Testing with Support for AES
+
+The AES AEAD variant `crypto_aead_aes256gcm` requires hardware support for the
+`AES` and `CLMUL` instruction set extensions to x86; you can read why that's the
+case
+[here](https://libsodium.gitbook.io/doc/secret-key_cryptography/aead/aes-256-gcm#limitations). These instruction set extensions were first made
+available in Intel Westmere (early 2010) and at the time of writing x86 hardware
+support for them is near universal. However, rustc is a cross compiler and will
+not enable them unless it's targeting a CPU that supports them; the baseline
+x86 target does not.
+
+If you need to use `crypto_aead_aes256gcm` you must build/test sodiumoxide with
+`RUSTFLAGS` set to include one of the following option sets:
+
+* `-C target-cpu=native` if you're building on a box with `AES` and `CLMUL`
+* `-C target-cpu=[target]` where `[target]` is `rustup` target with `AES` and
```

Comment:
I don't think this adds up, a `rustup` target, i.e. something listed by `rustup target list`, is a host-tuple that is passed via `--target` to rustc whereas the possible options for the target CPU are given by `rustc --print target-cpus`. (It is also not a target in itself but just an alias for a collection of target features AFAIK.)

Reply:
Yup, I didn't document that correctly. The new PR clarifies.